server/apps/publish/archive_publish.py

```diff
@@ -282,48 +302,92 @@ def publish(self, doc, updates, target_output_channels=None):
+
+                for subscriber in active_subscribers:
+                    found = False
+                    for s in subscribers:
+                        if s[config.ID_FIELD] == subscriber[config.ID_FIELD]:
+                            found = True
+                            break
+
+                    if not found:
+                        subscribers_yet_to_receive.append(subscriber)
+
+                if len(subscribers_yet_to_receive):
+                    formatters_not_found, queued_new_subscribers = \
+                        self.queue_transmission(updated, subscribers_yet_to_receive, target_media_type)
+                    no_formatters.extend(formatters_not_found)
+        elif self.published_state == 'published':  # Step 2(c)
+            # Step 2(c)(i)
+            subscribers = list(get_resource_service('subscribers').get(req=None, lookup={'is_active': True}))
+            no_formatters, queued = self.queue_transmission(updated, subscribers, target_media_type)
+
+        # Step 3
         user = get_user()
-
         if no_formatters and len(no_formatters) > 0:
             push_notification('item:publish:wrong:format',
-                              item=str(doc['_id']), unique_name=doc['unique_name'],
+                              item=str(doc[config.ID_FIELD]), unique_name=doc['unique_name'],
                               desk=str(doc.get('task', {}).get('desk', '')),
-                              user=str(user.get('_id', '')),
+                              user=str(user.get(config.ID_FIELD, '')),
                               formats=no_formatters)
 
-        if not target_output_channels and not queued:
+        if not target_media_type and not queued:
             raise PublishQueueError.item_not_queued_error(Exception('Nothing is saved to publish queue'), None)
 
         return queued
 
-    def queue_transmission(self, doc, target_output_channels=None):
+    def queue_transmission(self, doc, subscribers, target_media_type=None):
         """
-        Formats and queues the article to all active subscribers. Workflow:
-            1. Fetch Active Subscribers
-            2. For each Destination of a Subscriber,
-                a. format the article as per the format defined in the destination.
-                b. queue the formatted article
-        :return:
-            1. format list for which formatters are not found,
-            2. True if the article queued for transmissions False otherwise.
+        Method formats and then queues the article for transmission to the passed subscribers.
+        ::Important Note:: Format Type across Subscribers can repeat. But we can't have formatted item generated once
+        based on the format_types configured across for all the subscribers as the formatted item must have a published
+        sequence number generated by Subscriber.
+
+        :param: target_media_type - dictate if the doc being queued is a Takes Package or an Individual Article.
```

Comment:
Seems like we need to revise the convention then. But yeah, this would be out of the scope of this PR, thus the consistency argument prevails here, I agree.